base/math.jl

```diff
@@ -911,13 +911,19 @@ end
     z
 end
 @inline function ^(x::Float32, y::Float32)
-    z = ccall("llvm.pow.f32", llvmcall, Float32, (Float32, Float32), x, y)
-    if isnan(z) & !isnan(x+y)
-        throw_exp_domainerror(x)
-    end
-    z
+    x = Float64(x)
+    x<0 && !isinteger(y) && throw_exp_domainerror(x)
+    ans = Float32(exp2(log2(abs(x))*y))
+    x<0 && isodd(y) && return -ans
```

Comment:
Feels like just either

```
if x<0 && isodd(y)
    return -ans
else
    return ans
end
```

or

`return (x<0 && isodd(y)) ? -ans : ans`

is clearer.